Travel agency

KrasimirRaikov opened the pull request:
Two tables for people and their trips, and operations on the two tables

KrasimirRaikov commented:
made some changes

KrasimirRaikov commented:
I have made some design changes.

ddimitrov123 commented on src/com/clouway/jdbc/travel/agency/persistence/ClientTripsInfo.java:
```diff
@@ -0,0 +1,64 @@
+/**
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class ClientTripsInfo implements ClientTrips {
```
You can remove this class and move the metod in `PersistentTripRepository`

KrasimirRaikov replied:
moved it

ddimitrov123 commented on src/com/clouway/jdbc/travel/agency/persistence/ClientTripsInfo.java:
```diff
@@ -0,0 +1,64 @@
+        } finally {
+            if (statement != null) {
+                try {
+                    statement.close();
```
Why you close statement and resultset twice?

KrasimirRaikov replied:
I added the finally afterwards and forgot to remove the once in the try. I fixed it

ddimitrov123 commented on src/com/clouway/jdbc/travel/agency/persistence/PersistentClientRepository.java:
```diff
@@ -0,0 +1,151 @@
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
```
.

KrasimirRaikov replied:
fixed

ddimitrov123 commented on src/com/clouway/jdbc/travel/agency/persistence/PersistentTripRepository.java:
```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Override
+    public void book(Trip trip) {
```
I think register(Trip trip) fits better here

KrasimirRaikov replied:
fixed

ddimitrov123 commented on src/com/clouway/jdbc/travel/agency/persistence/PersistentTripRepository.java:
```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Override
+    public List<Trip> getList() {
```
findAll()

KrasimirRaikov replied:
fixed

KrasimirRaikov commented:
removed the `TravelAgency` class and replaced it's test with two classes for the two repositories, made the `id` of the `TripRepository` of type `long`.

panayotkulchev commented on src/com/clouway/jdbc/travel/agency/persistence/PersistentClientRepository.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    @Override
+    public List<Client> getWithNameBeggining(String nameBeginning) {
```
`nameBeginning` can be replaced with better name like `prefix`

panayotkulchev commented on src/com/clouway/jdbc/travel/agency/persistence/PersistentTripRepository.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Override
+    public List<Client> clientTripsOverlapBetween(Date startDate, Date endDate, String city) {
```
This method is like 30 lines without line space. I talk about code format. For better readability split in blocks bu functionalitiy with one empty line

panayotkulchev replied:
It can be applied to all your taskas in the PRs

panayotkulchev commented on test/com/clouway/jdbc/travel/agency/PersistentTripRepositoryTest.java:
```diff
@@ -0,0 +1,333 @@
+ * @author Krasimir Raikov(raikov.krasimir@gmail.com)
+ */
+public class PersistentTripRepositoryTest {
+    ClientRepository clientRepository = null;
```
Is it needed to assign `null` or it can be without it.

panayotkulchev commented on test/com/clouway/jdbc/travel/agency/PersistentTripRepositoryTest.java:
```diff
@@ -0,0 +1,333 @@
+
+        Trip tripActual = tripRepository.getById(1);
+        Trip anotherTripActual = tripRepository.getById(2);
+        assertThat(tripActual, is(equalTo(trip)));
```
Leave empty line before assertions so is easyer to read.

panayotkulchev commented on test/com/clouway/jdbc/travel/agency/PersistentTripRepositoryTest.java:
```diff
@@ -0,0 +1,333 @@
+        assertThat(tripRepository.citiesByPopularity(), is(equalTo(cities)));
+    }
+
+    private void registerToRepository(Client... people) {
```
pretendRegisteredPeopleAre() - it is common naming for such a method in our company

panayotkulchev replied:
pretendRegisteredClientsAre(Client ... client) not `peple`

panayotkulchev commented:
After you fix the notes you have LGTM
